internal/wasm/gofunc_test.go

```diff
@@ -133,3 +132,218 @@ func TestGetFunctionTypeErrors(t *testing.T) {
 		})
 	}
 }
+
+// stack simulates the value stack in a way easy to be tested.
```

Comment:
all the tests here are new as we never unit tested this logic before (it was covered, but tested indirectly)